File: py/entities/scalar.py
"""A scalar (time and value) entity."""

from datetime import datetime


class Scalar:
    """Represents a timestamped scalar value."""

    def __init__(self, time: datetime, value: float):
        self.time = time
        self.value = value

    def __repr__(self) -> str:
        return f"Scalar({self.time}, {self.value})"

File: rust/src/entities/scalar.rs
/// Represents a scalar value.
#[derive(Debug, Clone, Copy)]
pub struct Scalar {
    pub time: i64,
    pub value: f64,
}

impl Scalar {
    /// Creates a new Scalar.
    pub fn new(time: i64, value: f64) -> Self {
        Self { time, value }
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_new() {
        let s = Scalar::new(100, 3.14);
        assert_eq!(s.time, 100);
        assert_eq!(s.value, 3.14);
    }
}
